src/book/03-guides/04-general/04-documentation.mdx

```diff
@@ -0,0 +1,86 @@
+---
+section: Guides
+chapter: General
+title: Documentation
+description: How to properly document code.
+slug: /guides/general/documentation
+---
+
+## Conventions
+
+We use tokens called "directives" or "tags" to annotate different parts of a comment with rich metadata. These directives start with `@` or `\` (`@` is preferred) and are followed by the directive name, arguments (for some directives), and a single line of description.
+
+When writing modules, the `.hpp` should include a documenting comment for each class, struct, function, method, global variable, member variable and anything extra that you think may need one.
+
+### Directives
+
+These are some common directives:
+
+| Directive                         | Description                                                                                                                                                                                           |
+| --------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `@brief`                          | A brief description of the following code. Every documenting comment should have this. If the first line does not have a directive it is treated as `@brief`.                                         |
+| `@details`                        | All the details that someone using this thing might need to know beyond what is provided in other directives. Comment lines after the initial one but before any directive are treated as `@details`. |
```

Comment:
This raises a question for me. If I have a long brief which has a line break mid-text, will that automatically become part of the `details`? Could you please clarify this, or reword it so that it's obvious

Reply:
I think I describe it well, the first line is brief and all the other lines are details.

```
/**
 * This is brief
 * This is details
 * This still is details
 */
```